packages/lesswrong/components/hooks/useScreenWidth.ts

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from "react";
+import { isClient } from "../../lib/executionEnvironment";
+import { useTheme } from "../themes/useTheme";
+
+/**
+ * WARNING: This hook is not SSR safe!
+ *
+ * It assumes you're on the desktop and can cause layout shift on load for mobile
+ * users if you're not careful.
+ */
+export const useIsAboveScreenWidth = (targetScreenWidth: number) => {
+  const initialScreenWidth = isClient ? window.innerWidth : 4000;
+  const [actualScreenWidth, setActualScreenWidth] = useState(initialScreenWidth);
+  
+  useEffect(() => {
+    const handleResize = () => {
+      setActualScreenWidth(window.innerWidth);
+    }
+    window.addEventListener("resize", handleResize);
+    return () => window.removeEventListener("resize", handleResize);
+  }, [])
+  
+  if (!isClient) {
+    return true;
+  }
+  
+  
+  return actualScreenWidth > targetScreenWidth;
+}
+
+/**
+ * WARNING: This hook is not SSR safe!
+ *
+ * It assumes you're on the desktop and can cause layout shift on load for mobile
+ * users if you're not careful.
+ */
+export const useIsAboveBreakpoint = (breakpoint: BreakpointName) => {
+  const theme = useTheme();
+  const breakpointWidth = theme.breakpoints.values[breakpoint];
+  return useIsAboveScreenWidth(breakpointWidth);
+}
+
+/**
+ * WARNING: This hook is not SSR safe!
+ *
+ * It assumes you're on the desktop and can cause layout shift on load for mobile
+ * users if you're not careful.
+ */
+export const useIsDesktop = () => {
+  return useIsAboveBreakpoint('lg');
+}
+
+/**
+ * WARNING: This hook is not SSR safe!
+ *
+ * It assumes you're on the desktop and can cause layout shift on load for mobile
+ * users if you're not careful.
+ */
+export const useIsMobile = () => {
+  return !useIsAboveBreakpoint('sm');
+}
```

Comment:
So this is intentionally skipping the middle sizes? I think that's slightly surprising, so I would either write a comment about that, or add a `useIsTablet` to handle all cases

Reply:
It is. In my ontology, desktop is bigger than tablet and mobile is smaller, and I rarely am explicitly querying for isTablet. I'll make a note.